### PR TITLE
Update parent pom with required adjustments

### PR DESCRIPTION
**Pull request by Wadeck**
Update the pom to use the parent version 4.1 + some small refactor/cleanup

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,16 +3,15 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.2</version>
+    <version>4.1</version>
```
Comment (jglick) [inline pom.xml:6]:
```suggestion
    <version>4.4</version>
```

File: src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/ReverseProxySecurityRealm.java [hunk 2]
```diff
@@ -23,61 +23,23 @@
...
+import hudson.security.LDAPSecurityRealm;
+import hudson.security.SecurityRealm;
+import hudson.security.UserMayOrMayNotExistException;
 import hudson.tasks.Mailer;
 import hudson.tasks.Mailer.UserProperty;
 import hudson.util.FormValidation;
 import hudson.util.Scrambler;
 import hudson.util.spring.BeanBuilder;
-
-import java.io.File;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/ReverseProxySecurityRealm.java:42]: Maybe better to keep original simple lexicographic sort?

Comment (Wadeck) [reply src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/ReverseProxySecurityRealm.java]: A lot better now :) thanks for the catch

**Comment by Wadeck** [comment]
Supersed by https://github.com/jenkinsci/reverse-proxy-auth-plugin/pull/40

**Pull request closed without merging**